lib/Bridge/WorseReflection/Formatter/FunctionLikeSnippetFormatter.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Phpactor\Completion\Bridge\WorseReflection\Formatter;
+
+use Phpactor\Completion\Core\Formatter\Formatter;
+use Phpactor\Completion\Core\Formatter\ObjectFormatter;
+use Phpactor\Completion\Core\Util\Snippet\Placeholder;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionFunction;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionMethod;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionParameter;
+
+class FunctionLikeSnippetFormatter implements Formatter
+{
+    public function canFormat(object $function): bool
+    {
+        return $function instanceof ReflectionFunction
+            || $function instanceof ReflectionMethod;
+    }
+
+    public function format(ObjectFormatter $formatter, object $function): string
+    {
+        assert(
+            $function instanceof ReflectionFunction
+            || $function instanceof ReflectionMethod
+        );
+
+        $name = $function instanceof ReflectionFunction
+            ? $function->name()->short()
+            : $function->name()
+        ;
+        $parameters = $function->parameters();
+
+        if (0 === $parameters->count()) {
+            return "$name()";
```

Comment:
Shorter and clearer, for other places generally it's because one of the placeholder is an expression or because it will be too long on one line.
I generally avoid concatenation as possible and use both these forms to achieve that, but if it matters to you to be consistent there I will do it, just let me know.

Reply:
if only because this is the only place in the code that does it.